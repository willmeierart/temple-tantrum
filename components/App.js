// main wrapper component - layout, universal styles, etc.
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { checkIfMobile, getVPDims } from '../lib/redux/actions'
import Header from './core/Header'
import Footer from './core/Footer'

// import globalStyles from '../../styles/index.scss'

class App extends Component {
  componentDidMount () {

  }
  render () {
    const { children } = this.props
    return (
      <div className='app-outer'>
        <div className='app-inner'>
          <header>
            <Header />
          </header>
          <main>{ children }</main>
          <footer>
            <Footer />
          </footer>
        </div>
        <style jsx global>{`
          a {
            text-decoration: none;
            color: inherit;
          }
          li {
            list-style: none;
          }
          html, body {
            {/* overflow: hidden!important;
            position: fixed!important; */}
          }
          body {
            height: 100vh;
            width: 100vw;
            box-sizing: border-box;
          }
          header {}
          footer {}
          main {}
        `}</style>
        {/* <style dangerouslySetInnerHTML={{ __html: globalStyles }} /> */}
      </div>
    )
  }
}

function mapStateToProps (state) {
  return {
    isMobile: state.env.isMobile,
    dims: state.env.dims
  }
}

function mapDispatchToProps (dispatch) {
  return {
    onCheckIfMobile: () => dispatch(checkIfMobile()),
    onGetVPDims: () => dispatch(getVPDims())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(App)
// export default App

App.propTypes = {
  title: PropTypes.string.isRequired
}
