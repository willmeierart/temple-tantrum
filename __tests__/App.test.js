/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import App from '../components/App'

describe('With Enzyme', () => {
  it('renders correctly', () => {
    const app = shallow(<App />)
    expect(app).toMatchSnapshot()
  })
})

// describe('With Snapshot Testing', () => {
//   it('renders', () => {
//     const component = renderer.create(<App />)
//     const tree = component.toJSON()
//     expect(tree).toMatchSnapshot()
//   })
// })
