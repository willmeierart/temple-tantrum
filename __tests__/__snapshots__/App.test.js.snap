// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Enzyme renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CustomHead
          title={undefined}
        />,
        <div
          className="jsx-1733877071 app-inner"
        >
          <header
            className="jsx-1733877071"
          >
            <Header />
          </header>
          <main
            className="jsx-1733877071"
          />
          <footer
            className="jsx-1733877071"
          >
            <Footer />
          </footer>
        </div>,
        <JSXStyle
          css="a{text-decoration:none;color:inherit;}li{list-style:none;}body{height:100vh;width:100vw;box-sizing:border-box;}"
          styleId="1733877071"
        />,
      ],
      "className": "jsx-1733877071 app-outer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header
              className="jsx-1733877071"
            >
              <Header />
            </header>,
            <main
              className="jsx-1733877071"
            />,
            <footer
              className="jsx-1733877071"
            >
              <Footer />
            </footer>,
          ],
          "className": "jsx-1733877071 app-inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Header />,
              "className": "jsx-1733877071",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "jsx-1733877071",
            },
            "ref": null,
            "rendered": null,
            "type": "main",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Footer />,
              "className": "jsx-1733877071",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "footer",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": "a{text-decoration:none;color:inherit;}li{list-style:none;}body{height:100vh;width:100vw;box-sizing:border-box;}",
          "styleId": "1733877071",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CustomHead
            title={undefined}
          />,
          <div
            className="jsx-1733877071 app-inner"
          >
            <header
              className="jsx-1733877071"
            >
              <Header />
            </header>
            <main
              className="jsx-1733877071"
            />
            <footer
              className="jsx-1733877071"
            >
              <Footer />
            </footer>
          </div>,
          <JSXStyle
            css="a{text-decoration:none;color:inherit;}li{list-style:none;}body{height:100vh;width:100vw;box-sizing:border-box;}"
            styleId="1733877071"
          />,
        ],
        "className": "jsx-1733877071 app-outer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <header
                className="jsx-1733877071"
              >
                <Header />
              </header>,
              <main
                className="jsx-1733877071"
              />,
              <footer
                className="jsx-1733877071"
              >
                <Footer />
              </footer>,
            ],
            "className": "jsx-1733877071 app-inner",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Header />,
                "className": "jsx-1733877071",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "header",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "jsx-1733877071",
              },
              "ref": null,
              "rendered": null,
              "type": "main",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Footer />,
                "className": "jsx-1733877071",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "footer",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": "a{text-decoration:none;color:inherit;}li{list-style:none;}body{height:100vh;width:100vw;box-sizing:border-box;}",
            "styleId": "1733877071",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`With Snapshot Testing App shows "Hello world!" 1`] = `
<div
  data-jsx={2648947580}
>
  <p
    data-jsx={2648947580}
  >
    Hello World!
  </p>
</div>
`;
